### PR TITLE
[ctk] Add new port

### PR DESCRIPTION
**Pull request by raahilsha-z**
**Describe the pull request**

- #### What does your PR fix?
  Draft PR for adding CTK to vcpkg. This package has quite a few quirks, and I could use the eyes of someone more experienced with vcpkg to help resolve some of these. More details to follow in next comment.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Windows and Linux should be supported - still working on the port.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  I am still working on this PR

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/ctk/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+)
+
+# Configure and build
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DCTK_QT_VERSION=5
+        -DCTK_ENABLE_DICOM=ON
+        -DCTK_ENABLE_Widgets=ON
+        -DCTK_SUPERBUILD=ON
```
Comment (dg0yt) [inline ports/ctk/portfile.cmake:17]: This looks wrong. A superbuild usually pulls in third-party dependencies. But in vcpkg, dependencies shall come from vcpkg. 
Without the superbuild - with all dependencies pre-installed - you may perhaps easily switch to vcpkg's regular build functions.

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Comment by dg0yt** [comment]
The x64-windows build succeeded. But this is a problem:
~~~
CMake Warning at ports/qt5-base/portfile.cmake:27 (message):
  qt5-base is currently not setup to support feature 'mysqlplugin' on
  platforms other than windows.  Feel free to open up a PR to fix it!
~~~
I wonder if there is really a hard dependency on the `mysqlplugin`, or if it is just using independent Qt SQL API. If there is no hard dependency, the port shouldn't request `mysqlplugin` - it is up to the user to install the desired Qt features.

**Comment by raahilsha-z** [comment]
Sorry about that! Didn't realize that qt5-base contained Qt::Sql. Just pushed that change.

**Comment by raahilsha-z** [comment]
This builds fine, but unfortunately it seems like `vcpkg_cmake_config_fixup` isn't doing a thorough job/is breaking things. If I make a dummy project with the following (main.cpp is just a standard helloworld):

```
cmake_minimum_required(VERSION 2.8.6)
project(Example1)
find_package(CTK REQUIRED)
include(${CTK_USE_FILE})
add_executable(example1 main.cpp)
target_link_libraries(example1 ${CTK_LIBRARIES})
```

I get an error with the `find_package()` call:

```
CMake Error at C:/vcpkg/installed/x64-windows/share/ctk/CTKConfig.cmake:19 (message):
  File or directory  C:/vcpkg/installed/x64-windows/lib/ctk-0.1/CMake/UseCTK.cmake
  referenced by variable CTK_USE_FILE does not exist !
```

**Comment by dg0yt** [comment]
> it seems like `vcpkg_cmake_config_fixup` isn't doing a thorough job/is breaking things.

Well, CMake config can be almost everything. `vcpkg_cmake_config_fixup` is for the standard config generated by CMake - which doesn't have any "use" files or similar. You may need to add additonal fixup commands. (In the worst case, it cannot be used at all.)

**Comment by Adela0814** [comment]
Pinging @raahilsha-z  for response. Is work still being done for this PR?

**Comment by Adela0814** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**